### PR TITLE
!!! TASK: Rename object and resource

### PR DESCRIPTION
**Pull request by kdambekalns**
This renames the class `Resource` to `ResourceObject` and renames the namespaces
`TYPO3\Flow\Object` and `TYPO3\Flow\Resource` to `TYPO3\Flow\ObjectManagement`
and `TYPO3\Flow\ResourceManagement` respectively.

A Doctrine migration and two core migrations to help with adjusting code are added.

Resolves neos/flow-development-collection#719

### CODE REVIEW

**Comment by kdambekalns** [comment]
Please feel free to have a thorough look at this already!

Two notes: I did not touch the generated references, for they will be generated. And I still want to create a core migration, for use with Neos and for everyone out there.

File: TYPO3.Flow/Classes/TYPO3/Flow/Package.php [hunk 1]
```diff
@@ -86,14 +86,14 @@ public function boot(Core\Bootstrap $bootstrap)
                 return;
             }
             $objectManager = $bootstrap->getObjectManager();
-            $resourceManager = $objectManager->get(\TYPO3\Flow\Resource\ResourceManager::class);
+            $resourceManager = $objectManager->get(\TYPO3\Flow\ResourceManagement\ResourceManager::class);
```
Comment (daniellienert) [inline TYPO3.Flow/Classes/TYPO3/Flow/Package.php]: The namespaces should be relative (mhm .. thought I have done that already ....)

File: TYPO3.Flow/Classes/TYPO3/Flow/ResourceManagement/ResourceRepository.php [hunk 2]
```diff
@@ -157,8 +157,8 @@ public function findAllIterator()
         /** @var QueryBuilder $queryBuilder */
         $queryBuilder = $this->entityManager->createQueryBuilder();
         return $queryBuilder
-            ->select('Resource')
-            ->from($this->getEntityClassName(), 'Resource')
+            ->select('ResourceObject')
```
Comment (daniellienert) [inline TYPO3.Flow/Classes/TYPO3/Flow/ResourceManagement/ResourceRepository.php]: Seems the Database Migration is missing for this one.

Comment (kdambekalns) [inline TYPO3.Flow/Classes/TYPO3/Flow/ResourceManagement/ResourceRepository.php]: Good catch…

**Comment by kdambekalns** [comment]
Instructions: https://discuss.neos.io/t/renaming-flow-core-classes-resource-and-object/1200/9

**Pull request merged**